client/my-sites/patterns/components/pattern-gallery/client.tsx

```diff
@@ -14,12 +15,68 @@ import './style.scss';
 
 const LOGGED_OUT_USERS_CAN_COPY_COUNT = 3;
 
+type MasonryGalleryProps = PropsWithChildren< {
+	className?: string;
+	enableMasonry: boolean;
+} >;
+
+// Simulates a Masonry layout by applying negative `margin-top` on every item that doesn't sit in
+// the first row
+function MasonryGallery( { children, className, enableMasonry }: MasonryGalleryProps ) {
+	const ref = useRef< HTMLDivElement >( null );
+
+	useLayoutEffect( () => {
+		if ( ! ref.current || ! enableMasonry ) {
+			return;
+		}
+
+		const element = ref.current;
+
+		function calculateLayout() {
```

Comment:
Seems like we can move this function outside of the component. No need to redeclare it on every effect run.

Reply:
Sure 👍